### PR TITLE
Implement a util function for condition array and apply condition array to VMI

### PR DESCRIPTION
**Pull request by woohhan**
Closes: #15

### CODE REVIEW

File: pkg/util/condition_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	"time"
+)
+
+var _ = Describe("GetConditionByType", func() {
+	Context("if conditions has no matching conditionType", func() {
+		found, _ := GetConditionByType([]v1alpha1.Condition{
+			{
+				Type:               "type1",
+				Status:             corev1.ConditionTrue,
+				ObservedGeneration: 32,
```
Comment (hyoung-90) [inline pkg/util/condition_test.go:18]: Why do we need ObservedGeneration in the test code? It seems that the value is not used in the test code.

Comment (hbinkim) [reply pkg/util/condition_test.go:18]: As far as I understand it's an optional value suggested by [the proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions#noteworthy-choices) to identify outdated condition compare to the .metadata.generation value. Also `ObservedGeneration` can be used to distinguish values between intentionally setted zero ones versus unsetted ones.

Comment (woohhan) [reply pkg/util/condition_test.go:18]: We need to make sure that other values of condition are not changed when calling the function.

File: pkg/util/condition.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// SetConditionByType sets condition to conditions. If there is a matching condition.Type, update it, if not, add it. It Returns the new slice
+func SetConditionByType(conditions []v1alpha1.Condition, conditionType string, status corev1.ConditionStatus, reason, message string) []v1alpha1.Condition {
+	for i := range conditions {
+		if conditions[i].Type != conditionType {
+			continue
+		}
+		conditions[i].Status = status
+		conditions[i].Reason = reason
+		conditions[i].Message = message
+		conditions[i].LastTransitionTime = metav1.NewTime(time.Now())
```
[the remark this answers is not in the document]
Comment (woohhan) [reply pkg/util/condition.go]: Updated

**Comment by hbinkim** [comment]
LGTM

**Comment by hyoung-90** [comment]
LGTM

**Pull request merged**